### ISSUE TITLE
Get-CimInstance : Invalid class when starting RBM

### ISSUE THREAD

**Issue by Djago** [RainbowMiner/RainbowMiner#2138]
Hi!

A few days ago the system started with some errors in the start procedure (device detection and VM detection).
![image](https://user-images.githubusercontent.com/34612740/187553022-378a502f-563e-4eaa-a8d0-6303682490de.png)

Every time it tries to use Get-CimInstance I get:
Get-CimInstance : Invalid class
[debug_2022-08-30.zip](https://github.com/RainbowMiner/RainbowMiner/files/9456964/debug_2022-08-30.zip)

**Comment by RainbowMiner** [comment]
Wow, that's a weird one. But the mining runs just fine?

**Comment by RainbowMiner** [comment]
It seems like all Windows system calls fail. That also explains, why there is the system update failure warning. Has there been a Windows update recently? Maybe try to reinstall Powershell Core 7.2.6.

**Comment by RainbowMiner** [comment]
If the reinstall doesn't help, please try the following:
- hit the Windows Key and type "pwsh"
- now open the PowerShell 7 (x64) app
- copy and paste the following code lines, line by line and hit enter after each line, copy the results and paste them here.
 
```
Get-CimInstance Win32_OperatingSystem -Property LastBootUpTime
Get-CimInstance -ClassName CIM_Processor
Get-CimInstance -Class Win32_OperatingSystem -Property "TotalVisibleMemorySize","FreePhysicalMemory"
Get-CimInstance -class Win32_LogicalDisk -namespace "root\CIMV2"
```

**Comment by Djago** [comment]
> Wow, that's a weird one. But the mining runs just fine?

No, because RBM can't finish the benchmark in progress (because it's filled with different errors).


> It seems like all Windows system calls fail. That also explains, why there is the system update failure warning. Has there been a Windows update recently? Maybe try to reinstall Powershell Core 7.2.6. Have you installed the correct version (64bit)?
> 
> https://github.com/PowerShell/PowerShell/releases/download/v7.2.6/PowerShell-7.2.6-win-x64.msi

It wasn't installed at all. But after installing that, there was no change at all. Same errors.


> If the reinstall doesn't help, please try the following:
> 
> * hit the Windows Key and type "pwsh"
> * now open the PowerShell 7 (x64) app
> * copy and paste the following code lines, line by line and hit enter after each line, copy the results and paste them here.
> 
> ```
> Get-CimInstance Win32_OperatingSystem -Property LastBootUpTime
> Get-CimInstance -ClassName CIM_Processor
> Get-CimInstance -Class Win32_OperatingSystem -Property "TotalVisibleMemorySize","FreePhysicalMemory"
> Get-CimInstance -class Win32_LogicalDisk -namespace "root\CIMV2"
> ```

I'll try and come back later with the results!

**Comment by RainbowMiner** [comment]
It might be the case, that the WMI repository has been damaged. Try to rebuild it (but make a backup of that folder beforehand):
https://social.technet.microsoft.com/Forums/windowsserver/en-US/8c673b55-0db1-4ce1-bfaa-94527e8adb00/how-to-fix-quotinvalid-classquot-in-powershell?forum=winservergen

**Comment by RainbowMiner** [comment]
This pointed at a damaged WMI repository on the machine rather than at RainbowMiner itself. Closing as stale - please open a new issue with a current debug file if you still see this. Thanks!
